ArduPlane/altitude.cpp

```diff
@@ -679,16 +679,36 @@ void Plane::rangefinder_terrain_correction(float &height)
  */
 void Plane::rangefinder_height_update(void)
 {
-    float distance = rangefinder.distance_orient(ROTATION_PITCH_270);
-    
-    if ((rangefinder.status_orient(ROTATION_PITCH_270) == RangeFinder::Status::Good) && ahrs.home_is_set()) {
+    const auto orientation = rangefinder_orientation();
+    bool range_ok = rangefinder.status_orient(orientation) == RangeFinder::Status::Good;
+    float distance = rangefinder.distance_orient(orientation);
+    float corrected_distance = distance;
+
+    /*
+      correct distance for attitude
+     */
+    if (range_ok) {
+        // correct the range for attitude
+        const auto &dcm = ahrs.get_rotation_body_to_ned();
+
+        Vector3f v{1, 0, 0};
```

Comment:
I think if you put the corrected distance in here rather than multiply at the end it is a bit more intuitive. 
```suggestion
        Vector3f v{corrected_distance, 0, 0};
```
and/or a comment explaining the maths.

Reply:
i'm happy either way, I've changed it to do it the way you prefer